lib/reverse_words.rb

```diff
@@ -1,3 +1,7 @@
+# Share and explain the time and space complexities for your solution.
+# The time complexity is O(n) because it's two loops that are linear so it's O(n) + O(n)
+# The space complexity is O(1) because it's outputting the same amount as input
```

Comment:
The space complexity of your algorithm will be O(n). This is because you are creating a new array called word_indices to hold the indices. In the case where there is only word in the string, word_indices will be of size n. Without sufficient space to allocate this new array, your algorithm won't be able to execute and as the input string size grows, so will the size of word_indices grow linearly.
Slack me if you have questions on this.